beacon-chain/blockchain/receive_block.go

```diff
@@ -90,6 +93,9 @@ func (s *Service) ReceiveBlockBatch(ctx context.Context, blocks []interfaces.Rea
 	ctx, span := trace.StartSpan(ctx, "blockChain.ReceiveBlockBatch")
 	defer span.End()
 
+	s.cfg.ForkChoiceStore.Lock()
```

Comment:
Single global lock for all of batch processing. Need to check that `ReceiveBlock` is not called from within `ReceiveBlockBatch`